core/modules/widgets/action-clipboard.js

```diff
@@ -0,0 +1,83 @@
+/*\
+title: $:/core/modules/widgets/action-clipboard.js
+type: application/javascript
+module-type: widget
+
+Action widget to copy things to clipboard.
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+var Widget = require("$:/core/modules/widgets/widget.js").widget;
+
+var ClipboardWidget = function(parseTreeNode,options) {
+    this.initialise(parseTreeNode,options);
+};
+
+/*
+Inherit from the base widget class
+*/
+ClipboardWidget.prototype = new Widget();
+
+/*
+Render this widget into the DOM
+*/
+ClipboardWidget.prototype.render = function(parent,nextSibling) {
+    this.computeAttributes();
+    this.execute();
+};
+
+/*
+Compute the internal state of the widget
+*/
+ClipboardWidget.prototype.execute = function() {
+    this.actionTiddler = this.getAttribute("$tiddler", null);
+    this.copyText = this.getAttribute("$text", null);
+    this.tiddlerFormat = this.getAttribute("$format", null);
+};
+
+/*
+Refresh the widget by ensuring our attributes are up to date
+*/
+ClipboardWidget.prototype.refresh = function(changedTiddlers) {
+    var changedAttributes = this.computeAttributes();
+	if(Object.keys(changedAttributes).length) {
```

Comment:
TW has a utility function to count changes in an object `$tw.utils.count(changedAttributes)`, which is basically the same thing. But for consistency reasons it should be used that way. eg: 

https://github.com/Jermolene/TiddlyWiki5/blob/779ac28bd02defa158af470a1ce3e3f0125cba38/core/modules/widgets/let.js#L87 or https://github.com/Jermolene/TiddlyWiki5/blob/779ac28bd02defa158af470a1ce3e3f0125cba38/core/modules/widgets/element.js#L89-L91

Reply:
This line was directly stolen from another action widget, so if it's incorrect here it's incorrect there too.

Reply:
Should I fix this for the action widget it was stolen from as well?

Reply:
For consistency reasons I thin yes. Which widget was it?

Reply:
the action-log widget